source/lac/petsc_vector_base.cc

```diff
@@ -210,6 +210,27 @@ namespace PETScWrappers
 
 
 
+  void
+  VectorBase::clear ()
+  {
+    if (attained_ownership)
+      {
+#if DEAL_II_PETSC_VERSION_LT(3,2,0)
+        const int ierr = VecDestroy (vector);
+#else
+        const int ierr = VecDestroy (&vector);
```

Comment:
Sorry, I did not comment on my last change, I let the clear function in the base class but simplified clear in the derived classes a bit, you decide.